ui/src/components/LogsRow/LogsRow.tsx

```diff
@@ -34,23 +34,27 @@ export default function LogsRow({
   // Parses and displays the data at the time of the log.
   const fetchMetrics = async () => {
     try {
-      // Parse time into Prometheus API friendly format
-      const promTime = (Date.parse(time) / 1000).toFixed(3);
-
       // POST request to the backend via the ddClient.
       const response: any = await ddClient.extension.vm?.service?.get(
-        `/api/promQL?containerID=${containerId}&time=${promTime}`
+        `/api/promQL?containerID=${containerId}&time=${time.toString()}`
       );
 
       // If the returned value is a valid metric, show only up to two digits after the decimal.
       Object.keys(response).forEach((el) => {
-        if (response[el] !== 'NaN') {
+        if (response[el] !== 'No data') {
```

Comment:
I would slightly prefer to keep this at `NaN` instead of the string `no data` because `NaN` is a number type and we can add type safety in the backend.

Reply:
The "No data" string is somewhat arbitary and is repeated 3 times in this file. A typo could result in a bug